tests/opt-testcases/tls13-kex-modes.sh

```diff
@@ -993,7 +993,7 @@ requires_config_enabled MBEDTLS_DEBUG_C
 requires_config_enabled MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED
 run_test    "TLS 1.3: O->m: ephemeral_all/psk, fail, no common kex mode" \
             "$P_SRV tls13_kex_modes=psk debug_level=5 $(get_srv_psk_list)" \
-            "$O_NEXT_CLI -tls1_3 -msg   \
+            "$O_NEXT_CLI_NO_PURE_PSK -tls1_3 -msg   \
```

Comment:
Not a big fan of changing the default options in `O_NEXT` and having to change so many existing tests as a consequence.

Also, why do you feel the need to do this? The defragmentation tests were passing without it. I strongly suspect it's because you removed the `requires_certificate_authentication` in front of the "no fragmentation, for reference" test cases. If so, I'd much prefer if you just re-introduced this dependency there. We don't need to have those cases pass in a wider range of configurations than the cases for which they provide a reference point.

On a general note, let's try to remain focused on the task at hand and avoid larger changes until defragmentation is in a good shape.

Reply:
As discussed elsewhere, I've backed out this change, but I do want to reduce the places where we require a certificate, because I do want to have these tests running with PSK, at least with a small fragment size. It doesn't have to be right now in this pull request, but it would have to be in the next few days anyway, so we might as well take whatever we have that passes the CI and runs more tests.

Reply:
Filed as https://github.com/Mbed-TLS/mbedtls/issues/10033